lib/_http_client.js

```diff
@@ -448,6 +455,7 @@ function responseOnEnd() {
     }
     socket.removeListener('close', socketCloseListener);
     socket.removeListener('error', socketErrorListener);
+    socket.once('error', freeSocketErrorListener);
     // Mark this socket as available, AFTER user-added end
     // handlers have a chance to run.
     process.nextTick(emitFreeNT, socket);
```

Comment:
After push free socket to freeSockets https://github.com/nodejs/node/blob/master/lib/_http_agent.js#L77 , then add the `freeSocketErrorListener` to handle the errors.

```diff
// lib/_http_agent.js#L77

socket._httpMessage = null;
self.removeSocket(socket, options);
freeSockets.push(socket);
+++ socket.once('error', freeSocketErrorListener);
```